content/docs/tasks/traffic-management/secure-ingress.md

```diff
@@ -0,0 +1,285 @@
+---
+title: Secure Ingress Traffic Control
+description: Describes how to configure Istio to expose a service outside of the service mesh, over TLS or Mutual TLS.
+weight: 31
+keywords: [traffic-management,ingress]
+---
+
+> Note: This task uses the new [v1alpha3 traffic management API](/blog/2018/v1alpha3-routing/). The old API has been deprecated and will be removed in the next Istio release. If you need to use the old version, follow the docs [here](https://archive.istio.io/v0.7/docs/tasks/traffic-management/).
+
+The [Control Ingress Traffic](/docs/tasks/traffic-management/ingress.md) task describes how to configure an ingress
+gateway to enable TLS (non-mutual) traffic from outside the mesh into the mesh. This task extends that task to enable
+Mutual TLS ingress traffic.
+
+## Before you begin
+
+Perform the steps in the [Before you begin](/docs/tasks/traffic-management/ingress.md#before-you-begin) and [Determining the ingress IP and ports](/docs/tasks/traffic-management/ingress/#determining-the-ingress-ip-and-ports) sections of the
+[Control Ingress Traffic](/docs/tasks/traffic-management/ingress.md) task. After performing those steps you should have Istio and _httpbin_ service deployed, and the environment variables `INGRESS_HOST` and `SECURE_INGRESS_PORT` set.
+
+## Generate client and server certificates and keys
+
+For this task you can use your favorite tool to generate certificates and keys. We used [a script](https://github.com/nicholasjackson/mtls-go-example/blob/master/generate.sh)
+from the https://github.com/nicholasjackson/mtls-go-example repository.
+
+1.  Clone the https://github.com/nicholasjackson/mtls-go-example repository:
+
+    ```command
+    $ git clone https://github.com/nicholasjackson/mtls-go-example
+    ```
+
+1.  Change directory to the cloned repository:
+
+    ```command
+    $ cd mtls-go-example
+    ```
+
+1.  Generate the certificates (use any password):
+
+    ```command
+    generate.sh httpbin.example.com <password>
+    ```
+
+    The command will generate four directories: `1_root`, `2_intermediate`, `3_application` and `4_client` with client
+    and server certificates you will use.
+
+## Configure a TLS ingress gateway
+
+In this subsection we configure an ingress gateway with the port 443 to handle HTTPS traffic. We create a secret
+with a certificate and a private key. Then we create a `Gateway` definition that contains a `server` on the port 443.
+
+1. Create a Kubernetes `Secret` to hold the server's certificate and private key. Use `kubectl` to create the secret
+`istio-ingressgateway-certs` in namespace `istio-system` . The Istio gateway will load the secret automatically.
+
+    > The secret MUST be called `istio-ingressgateway-certs` in the `istio-system` namespace, or it will not
```

Comment:
@geeknoid The issue is that this task involves several risky points, this is why there are so many notes. Each note relates to a possible user mistake that can cause Istio ingress to break. There are two notes about using a wrong name of a certificate, two notes about using a wrong location of a certificate, and two notes about waiting for the gateway definition propagation, otherwise the users could be confused by unexpected results. The task itself is a bit tricky.